cpp/include/cuspatial/soa_readers.hpp

```diff
@@ -16,63 +16,67 @@
  **/
-gdf_column read_uint32_soa(const char *filename);
+std::unique_ptr<cudf::column> read_uint32_soa(const char *filename, rmm::mr::device_memory_resource* mr);
 
 /**
  * @brief read timestamp data from file as column
  *
  * @param[in] filename: file to read
  *
- * @return gdf_column storing its_timestamp data
+ * @return column storing its_timestamp data
 **/
-gdf_column read_timestamp_soa(const char *filename);
+std::unique_ptr<cudf::column> read_timestamp_soa(const char *filename, rmm::mr::device_memory_resource* mr);
 
 /**
- * @brief read lon/lat from file as two columns; data type is fixed to double (GDF_FLOAT64)
+ * @brief read lon/lat from file as two columns; data type is fixed to FLOAT64
  *
  * @param[in] filename: file name of point data in location_3d layout (lon/lat/alt but alt is omitted)
  *
- * @return gdf_columns storing x and y data
+ * @return columns storing x and y data
 **/
-std::pair<gdf_column, gdf_column>  read_lonlat_points_soa(const char *filename);
+std::pair<std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>>
+read_lonlat_points_soa(const char *filename, rmm::mr::device_memory_resource* mr);
 
 /**
- * @brief read x/y from file as two columns; data type is fixed to double (GDF_FLOAT64)
+ * @brief read x/y from file as two columns; data type is fixed to FLOAT64
  * 
  * @param[in] filename: file name of point data in coordinate_2d layout (x/y)
  * 
- * @return gdf_columns storing x and y data
+ * @return columns storing x and y data
 **/
-std::pair<gdf_column, gdf_column>  read_xy_points_soa(const char *filename);
+std::pair<std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>>
+read_xy_points_soa(const char *filename, rmm::mr::device_memory_resource* mr);
 
 /**
  * @brief read polygon data from file in SoA format
  * 
- * data type of vertices is fixed to double (GDF_FLOAT64)
+ * data type of vertices is fixed to FLOAT64
  *
  * @param[in] filename: polygon data filename
- * @param[out] ply_fpos: index polygons: prefix sum of number of rings of all
- *             polygons
- * @param[out] ply_rpos: index rings: prefix sum of number of vertices of all
- *             rings
- * @param[out] ply_x: x coordinates of concatenated polygons
- * @param[out] ply_y: y coordinates of concatenated polygons
  *
  * @note: x/y can be lon/lat.
+ *
+ * @return: vector of columns
+ *          column(0): index polygons: prefix sum of number of rings of all
+ *                     polygons
+ *          column(1): index rings: prefix sum of number of vertices of all
+ *                     rings
+ *          column(2): x coordinates of concatenated polygons
+ *          column(3): y coordinates of concatenated polygons
 **/
-void read_polygon_soa(const char *filename,
-                      gdf_column* ply_fpos, gdf_column* ply_rpos,
-                      gdf_column* ply_x, gdf_column* ply_y);
+std::vector<cudf::column> read_polygon_soa(const char *filename, rmm::mr::device_memory_resource* mr);
```

Comment:
Need unique_ptrs in the output...

```suggestion
std::vector<cudf::column>
read_polygons(std::string filepath, 
              rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
```